contracts/BandOracle.cdc

```diff
@@ -378,7 +378,7 @@ pub contract BandOracle {
     /// @param payment: Flow token vault containing the service fee.
     /// @return The `ReferenceData` containing the requested data.
     ///
-    pub fun getReferenceData (baseSymbol: String, quoteSymbol: String, payment: @FungibleToken.Vault): ReferenceData {
+    access(all) fun getReferenceData (baseSymbol: String, quoteSymbol: String, payment: @{FungibleToken.Vault}): ReferenceData {
         pre {
             BandOracle._getRefData(symbol: baseSymbol) != nil: "No quotes for base symbol"
```

Comment:
This is a pain point for the update, I'm thinking about possible solutions here. @sisyphusSmiling do you think it will be ok to change the function to return `ReferenceData?` rather than `ReferenceData`, and then move the checks on the pre to the body of the method? I still need to double check that returning an optional won't affect the rest of the contract